cmd/block-generator/runner/run.go

```diff
@@ -0,0 +1,260 @@
+	if err := indexerShutdownFunc(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Helper to record the import rate.
+func recordDataToFile(entry Entry, key string, out *os.File) error {
+	sum := 0.0
+	count := 0.0
+
+	for _, metric := range entry.Data {
+		var err error
+		if strings.HasPrefix(metric, "indexer_daemon_import_time_sec_sum") {
+			val := strings.Split(metric, " ")[1]
+			sum, err = strconv.ParseFloat(val, 64)
+		}
+		if strings.HasPrefix(metric, "indexer_daemon_import_time_sec_count") {
+			val := strings.Split(metric, " ")[1]
+			count, err = strconv.ParseFloat(val, 64)
+		}
+		if err != nil {
+			return fmt.Errorf("unable to parse metric '%s': %w", metric, err)
+		}
+	}
+	rate := sum / count
+
+	msg := fmt.Sprintf("%s:%f\n", key, rate)
+	if _, err := out.WriteString(msg); err != nil {
+		return fmt.Errorf("unable to write metric '%s': %w", key, err)
+	}
+	return nil
+}
+
+// Run the test for 'RunDuration', collect metrics and write them to the 'ReportDirectory'
+func (r *Args) runTest(indexerURL string, generatorURL string) error {
+	collector := &MetricsCollector{MetricsURL: fmt.Sprintf("http://%s/metrics", indexerURL)}
+
+	baseName := filepath.Base(r.Path)
+	baseNameNoExt := strings.TrimSuffix(baseName, filepath.Ext(baseName))
+	reportPath := path.Join(r.ReportDirectory, fmt.Sprintf("%s.report", baseNameNoExt))
+
+	report, err := os.Create(reportPath)
+	if err != nil {
+		return fmt.Errorf("unable to create report: %w", err)
+	}
+	defer report.Close()
+
+	// Run for r.RunDuration
+	start := time.Now()
+	for time.Since(start) < r.RunDuration {
+		time.Sleep(r.RunDuration / 10)
+		if err := collector.Collect("indexer_daemon_import_time_sec"); err != nil {
+			return fmt.Errorf("problem collecting metrics: %w", err)
+		}
+	}
+	if err := collector.Collect("indexer_daemon_import_time_sec"); err != nil {
+		return fmt.Errorf("problem collecting metrics: %w", err)
+	}
+
+	// Collect results.
+	durationStr := fmt.Sprintf("test_duration_seconds:%d\n", uint64(r.RunDuration.Seconds()))
+	if _, err := report.WriteString(durationStr); err != nil {
+		return fmt.Errorf("unable to write duration metric: %w", err)
+	}
+
+	resp, err := http.Get(fmt.Sprintf("http://%s/report", generatorURL))
+	if err != nil {
+		return fmt.Errorf("the process failed to start properly, health endpoint query failed")
+	}
+	defer resp.Body.Close()
+	var generatorReport generator.Report
+	if err := json.NewDecoder(resp.Body).Decode(&generatorReport); err != nil {
+		return fmt.Errorf("problem decoding generator report: %w", err)
+	}
+	for metric, entry := range generatorReport {
+		// Skip this one
+		if metric == "genesis" {
+			continue
+		}
+		str := fmt.Sprintf("transaction_%s_total:%d\n", metric, entry.GenerationCount)
+		if _, err := report.WriteString(str); err != nil {
+			return fmt.Errorf("unable to write transaction_count metric: %w", err)
+		}
+	}
+
+	// Record a rate from one of the first data points.
+	if len(collector.Data) > 5 {
```

Comment:
No, the metrics are aggregated. When we call `importTimeHistogramSeconds.Observe(dt.Seconds())` it increments a counter and adds `dt.Seconds()` to a sum.

In this code I happen to be collecting the aggregated metrics 11 times during the course of the test, but if we wanted graphs it would be better to hook up a Prometheus DB and query it directly.